projectq/meta/_compute.py

```diff
@@ -87,6 +87,21 @@ def _add_uncompute_tag(self, cmd):
         cmd.tags.append(UncomputeTag())
         return cmd
 
+    def _send_rewritten_command(self, cmd, rewrite_id_map):
```

Comment:
Extracting small private functions is a very common and widely recommended refactoring.

Extracting functions divides the problem into manageable pieces. It limits the amount of shared context between code in the main body. It reduces various measures of complexity, such as cyclomatic complexity. It allows the possibility of separately unit-testing the behavior of the small function. It creates function names that play a role similar to comments.

In my experience interviewing candidates, whether or not they approach a sub-problem by inlining it into their code or by delegating to a function is a strong predictor of whether they will finish on time.

For an online discussion of function lengths, see the software-se question ["What is the ideal length of a method for you?"](https://softwareengineering.stackexchange.com/questions/133404/what-is-the-ideal-length-of-a-method-for-you).